src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -403,31 +403,80 @@ private static string GetParameterText(string parameterName)
 
                 CommandProcessorBase commandProcessor = null;
                 CommandRedirection[] commandRedirection = null;
-                for (int i = 0; i < pipeElements.Length; i++)
-                {
-                    commandRedirection = commandRedirections != null ? commandRedirections[i] : null;
-                    commandProcessor = AddCommand(pipelineProcessor, pipeElements[i], pipeElementAsts[i],
-                                                  commandRedirection, context);
-                }
 
-                var cmdletInfo = commandProcessor?.CommandInfo as CmdletInfo;
-                if (cmdletInfo?.ImplementingType == typeof(OutNullCommand))
+                var pipelineAst = (PipelineAst)pipeElementAsts[0].Parent;
+                if (pipelineAst.BackgroundProcess)
+                {
+                    // For background jobs rewrite the pipeline as a Start-Job command
+                    var scriptblockBodyString = ((PipelineAst)pipeElementAsts[0].Parent).Extent.Text;
+                    var pipelineOffset = pipelineAst.Extent.StartOffset;
+                    var variables = pipelineAst.FindAll(x => x is VariableExpressionAst, true);
+                    // Used to make sure that the job runs in the current directory
+                    const string cmdPrefix = @"Microsoft.PowerShell.Management\Set-Location $using:pwd; ";
+                    // Minimize allocations by initializing the stringbuilder to the size of the source string + prefix + space for ${using:} * 2
+                    System.Text.StringBuilder updatedScriptblock = new System.Text.StringBuilder(cmdPrefix.Length + scriptblockBodyString.Length + 18);
+                    updatedScriptblock.Append(cmdPrefix);
+                    int position = 0;
+                    // Prefix variables in the scriptblock with $using: 
+                    foreach (var v in variables)
+                    {
+                        var vName = ((VariableExpressionAst) v).VariablePath.UserPath;
+                        // Skip variables that don't exist
+                        if (funcContext._executionContext.EngineSessionState.GetVariable(vName) == null)
+                            continue;
+                        // Skip PowerShell magic variables
+                        if (Regex.Match(vName,
+                                "^(global:){0,1}(PID|PSVersionTable|PSEdition|PSHOME|HOST|TRUE|FALSE|NULL)$", 
+                                    RegexOptions.IgnoreCase|RegexOptions.CultureInvariant).Success == false
+                        )
+                        {
+                            updatedScriptblock.Append(scriptblockBodyString.Substring(position, v.Extent.StartOffset - pipelineOffset - position));
+                            updatedScriptblock.Append("${using:");
+                            updatedScriptblock.Append(vName);
```

Comment:
Annoying corner case - you need to escape `}` in vName. `CodeGeneration.EscapeVariableName` will handle that for you.